### PR TITLE
Onboarding fixes

### PR DESCRIPTION
**Pull request by rezkiy37**
### Details

The PR fixes and improves the onboarding flow. The main changes:

1. Fixed modal background of the pages. It is translucent for web and desktop right now.
2. Fixed replies count.
3. Use bold text following the design.
4. Integrated `failureData` for the **CompleteGuidedSetup** command.
5. Updated a translation.

### Fixed Issues

$ https://github.com/Expensify/App/issues/38771
$ https://github.com/Expensify/App/issues/40725
PROPOSAL: N/A

### Tests

**Note:** To test with an existing user, you can initiate the flow from **Settings > Troubleshoot > Onboarding Flow**.

#### With the creation of a workspace (_Track business expenses..., Manage my team’s..._)

1. Open the app with an entire new user.
2. Verify that the "What do you want to do today?" modal appears.
3. Select any tutorial.
4. Click on "Continue".
5. Verify that the "What’s your name?" modal appears.
6. Enter any names.
7. Click on "Continue".
8. Verify that the "Where do you work?" modal appears.
10. Enter any name.
11. Click on "Continue".
12. Verify that the app redirects to the home page. 
13. Open the Concierge chat.
14. Verify that the app has generated the proper messages.
15. Verify that the app has created a workspace with the previously entered name. 
16. Verify that attachment messages are replaced with videos, once the backend sends an update.

#### Without the creation of a workspace (_others_)

1. Open the app with an entire new user.
2. Verify that the "What do you want to do today?" modal appears.
3. Select any tutorial.
4. Click on "Continue".
5. Verify that the "What’s your name?" modal appears.
6. Enter any names.
7. Click on "Continue".
8. Verify that the app redirects to the home page. 
9. Open the Concierge chat.
10. Verify that the app has generated the proper messages.
11. Verify that attachment messages are replaced with videos, once the backend sends an update.

#### Fixes:

1. Fixed modal background of the pages. It is translucent for web and desktop right now.
a. Verify that the background behind modal screens is translucent.

2. Fixed replies count.
a. Verify that the app optimistically shows 1 replay count for the "Meet you specialist" task of the "Manage my team's..." purpose.

4. Use bold text following the design.
a. Verify that the app optimistically generates bold texts as required: https://github.com/Expensify/App/pull/39687#issuecomment-2069045330, https://github.com/Expensify/App/pull/39687#issuecomment-2069038305.

6. Integrated `failureData` for the **CompleteGuidedSetup** command.
a. Enable the "Simulate failing network requests" setting. 
b. Verify that the app sets errors for messages in the Concierge tasks. Also, the app deletes task reports and resets the onboarding choice.

8. Updated a translation.
a. Verify that content is translated properly on the onboarding pages.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as **"Tests"**.

### QA Steps

Same as **"Tests"**.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/57314631/db1a135c-2e7f-414b-ac6d-a2bd3f2e1436

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/57314631/4f3519ac-37c3-4fb5-9e2c-f71e3e48c423

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/57314631/c944ebf5-e34d-418f-aa64-9175d63899d5

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

###### Translations 

Screenshot 2024-04-23 at 14 16 40
Screenshot 2024-04-23 at 14 16 45
Screenshot 2024-04-23 at 14 16 49

###### Integrated `failureData` for the **CompleteGuidedSetup** command.

Screenshot 2024-04-23 at 14 39 59

###### Bold texts 

Screenshot 2024-04-22 at 13 46 56
Screenshot 2024-04-22 at 13 53 15
Screenshot 2024-04-22 at 13 53 22
Screenshot 2024-04-22 at 13 53 43
Screenshot 2024-04-22 at 13 53 47
Screenshot 2024-04-22 at 13 53 57
Screenshot 2024-04-22 at 13 54 14
Screenshot 2024-04-22 at 13 54 19
Screenshot 2024-04-22 at 13 54 31
Screenshot 2024-04-22 at 13 55 09
Screenshot 2024-04-22 at 13 55 18
Screenshot 2024-04-22 at 13 55 25
Screenshot 2024-04-22 at 13 55 37
Screenshot 2024-04-22 at 13 55 41
Screenshot 2024-04-22 at 13 55 53
Screenshot 2024-04-22 at 13 55 58
Screenshot 2024-04-22 at 13 56 06
Screenshot 2024-04-22 at 13 56 14
Screenshot 2024-04-22 at 13 56 26

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/57314631/e8a7cfc4-df5e-43fa-8654-7ace36d2d681

</details>

### CODE REVIEW

**Comment by rushatgabhane** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
Bug: App crash when proceeding with "Something else" in onboarding flow

1. Start onboarding flow from troubleshoot menu
2. Select something else
3. Add name, go next
4. App crash

https://github.com/Expensify/App/assets/29673073/4fa00f2f-75f2-44c2-bc90-b2fed6ab584d

@rezkiy37 are you able to reproduce this?

**Comment by rushatgabhane** [comment]
it's crashing for all flows actually.


https://github.com/Expensify/App/assets/29673073/9466868c-5f34-45bd-91fe-95f12543c396

**Comment by trjExpensify** [comment]
Is it just on iOS native, @rushatgabhane?

**Comment by mountiny** [comment]
@rushatgabhane do you have any errors in console in the mweb flow?

**Comment by trjExpensify** [comment]
@rezkiy37 is this PR supposed to include the fix for the error when completing a task from the DM chat before clicking into the taskReport? CC: @thienlnam

**Comment by thienlnam** [comment]
I forgot where we landed on that - but yes let me know if there's an additional change that is needed to be made on my end

**Comment by trjExpensify** [comment]
Pretty sure [this is where we landed](https://expensify.slack.com/archives/C036QM0SLJK/p1713181178174279?thread_ts=1712620115.124819&cid=C036QM0SLJK), to share the task with the participant of the Concierge/SystemDM. 

It's just a little unclear if Michael is actioning on that here (for me at least!).

**Comment by trjExpensify** [comment]
👋 @rezkiy37 can you make sure to update [this as well](https://github.com/Expensify/App/blob/fd8ef990c108ed939cd6a1fd49027beb11e0570b/src/languages/en.ts#L1319-L1323), to the suggested copy by @dubielzyk-expensify [here](https://github.com/Expensify/App/pull/39687#issuecomment-2071113046). Thanks!

**Comment by rushatgabhane** [comment]
yes this is iOS native only

**Comment by rezkiy37** [comment]
> 👋 @rezkiy37 can you make sure to update [this as well](https://github.com/Expensify/App/blob/fd8ef990c108ed939cd6a1fd49027beb11e0570b/src/languages/en.ts#L1319-L1323), to the suggested copy by @dubielzyk-expensify [here](https://github.com/Expensify/App/pull/39687#issuecomment-2071113046). Thanks!

Updated - c3f41b3. Asked in [Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1713952257173539).

cc @dubielzyk-expensify 

<details><summary>Details</summary>
<p>

Screenshot 2024-04-24 at 11 47 33


</p>
</details>

**Comment by rezkiy37** [comment]
> it's crashing for all flows actually.
> 
>  Screen.Recording.2024-04-24.at.01.20.47.mov

I've found the root cause. I think it happens because of a pretty fast navigation action in timeout. I increased the delay to open the modal. It still looks smooth and solid. WDYT?
Fixed - 6d6a26e.

<details><summary>Details</summary>
<p>

https://github.com/Expensify/App/assets/57314631/bb2db4fd-a4ac-46e4-b057-63616852c624


https://github.com/Expensify/App/assets/57314631/1fae8edd-c77a-45ae-917c-68704f743a7b

</p>
</details> 

cc @trjExpensify @rushatgabhane @mountiny

File: src/languages/en.ts [hunk 1]
```diff
@@ -1319,8 +1319,7 @@ export default {
     onboarding: {
         welcomeVideo: {
             title: 'Welcome to Expensify',
-            description: 'Getting paid is as easy as sending a message.',
-            button: "Let's go",
+            description: 'One app to handle all your business and personal spend in a chat. Built for your business, your team, and your friends.',
```
Comment (mountiny) [inline src/languages/en.ts:1322]: Waiting for the [translations](https://expensify.slack.com/archives/C01GTK53T8Q/p1713952257173539)

**Comment by rezkiy37** [comment]
> @rezkiy37 is this PR supposed to include the fix for the error when completing a task from the DM chat before clicking into the taskReport? CC: @thienlnam

I tested this case. The backend does not accept completing by clicking on task reports. The app passed all IDs correctly. I can complete tasks, once I open task reports. I am not sure that it is a frontend bug. The backend successfully opens task reports, but not complete. 

<details><summary>Details</summary>
<p>

https://github.com/Expensify/App/assets/57314631/932251bc-6336-45ec-be80-c6fae3dde6e6

https://github.com/Expensify/App/assets/57314631/da85c3b3-587b-4be5-ab73-0236d3efc02e

</p>
</details> 

cc @trjExpensify @thienlnam

File: src/languages/es.ts [hunk 2]
```diff
@@ -1318,8 +1318,7 @@ export default {
     onboarding: {
         welcomeVideo: {
             title: 'Bienvenido a Expensify',
-            description: 'Cobrar es tan fácil como enviar un mensaje.',
-            button: 'Vámonos',
+            description: 'Una aplicación para gestionar todos tus gastos empresariales y personales en un chat. Pensada para tu empresa, tu equipo y tus amigos.',
```
Comment (pecanoro) [inline src/languages/es.ts:1321]:
```suggestion
            description: 'Una aplicación para gestionar todos tus gastos de empresa y personales en un chat. Pensada para tu empresa, tu equipo y tus amigos.',
```

**Review by pecanoro — CHANGES_REQUESTED** [review]

File: src/styles/variables.ts [hunk 3]
```diff
@@ -203,7 +203,7 @@ export default {
     sectionIllustrationHeight: 220,
     photoUploadPopoverWidth: 335,
     onboardingModalWidth: 500,
-    welcomeVideoDelay: 500,
+    welcomeVideoDelay: 1000,
```
Comment (mountiny) [inline src/styles/variables.ts:206]: It does not feel very reliable way here but I guess its ok for now. If the device is slower I wonder if the crash couls still occur

**Review by mountiny — DISMISSED** [review]
other than the translations, changes look good to me

File: src/styles/variables.ts (hunk 3 above)
Comment (rezkiy37) [reply src/styles/variables.ts:206]: Hmm... I tried it on some Android devices. It works fine even with 500. So we can try to fix it better later.

File: src/languages/es.ts (hunk 2 above)
Comment (rezkiy37) [reply src/languages/es.ts]: Done - 9968865.

File: src/languages/en.ts (hunk 1 above)
Comment (rezkiy37) [reply src/languages/en.ts:1322]: Done - 9968865.

File: src/styles/variables.ts (hunk 3 above)
Comment (mountiny) [reply src/styles/variables.ts:206]: Yeah definitely nothing to worry about now, let see if we will notice this issue in future

**Review by rushatgabhane — COMMENTED** [review]
Should the popover be over a blank screen? It should be over the home screen, right?



Screenshot 2024-04-24 at 16 39 43

https://github.com/Expensify/App/assets/29673073/b9413b24-2457-4fcd-9e2a-ebe65a25ce35

**Comment by trjExpensify** [comment]
>I tested this case. The backend does not accept completing by clicking on task reports. The app passed all IDs correctly. I can complete tasks, once I open task reports. I am not sure that it is a frontend bug. The backend successfully opens task reports, but not complete. Is it possible that the backend does not see task reports until I open it?

On this, @thienlnam's explanation [in thread](https://expensify.slack.com/archives/C036QM0SLJK/p1712961069852729?thread_ts=1712620115.124819&cid=C036QM0SLJK) is that until you open a task, you aren't granted ‘read, write, share’ permissions. So if you try to complete the task from the "preview" in the parent chat before you've opened the taskReport up, it fails due to a lack of permissions.

It sounds like we need to fix this on the backend then. It's pretty strange to show a preview and allow an action to be taken within said preview that will always result in an error. You can take actions to like [Approve]/[Pay] an expenseReport from it's preview component in the parent chat without having to open the expenseReport first, so I think we should bring taskReports in-line to allow the same. @thienlnam will you progress this?

**Comment by rushatgabhane** [comment]
Yes, it says you don't have permission but you can dismiss the error and the task will be marked as complete.

image

**Review by rushatgabhane — APPROVED** [review]
Other than https://github.com/Expensify/App/pull/40688#pullrequestreview-2019569990, PR LGTM! 🎉

**Comment by trjExpensify** [comment]
> > I tested this case. The backend does not accept completing by clicking on task reports. The app passed all IDs correctly. I can complete tasks, once I open task reports. I am not sure that it is a frontend bug. The backend successfully opens task reports, but not complete. Is it possible that the backend does not see task reports until I open it?
> 
> On this, @thienlnam's explanation [in thread](https://expensify.slack.com/archives/C036QM0SLJK/p1712961069852729?thread_ts=1712620115.124819&cid=C036QM0SLJK) is that until you open a task, you aren't granted ‘read, write, share’ permissions. So if you try to complete the task from the "preview" in the parent chat before you've opened the taskReport up, it fails due to a lack of permissions.
> 
> It sounds like we need to fix this on the backend then. It's pretty strange to show a preview and allow an action to be taken within said preview that will always result in an error. You can take actions to like [Approve]/[Pay] an expenseReport from it's preview component in the parent chat without having to open the expenseReport first, so I think we should bring taskReports in-line to allow the same. @thienlnam will you progress this?

Follow-up on this. It looks like for a task in a "normal" DM, we don't let the user get that far until they've clicked into the taskReport first. We disable the action from being taken until then:

image

So another option as an interim step might be to do the same in the Concierge DM until we figure out a broader plan.

**Comment by shawnborton** [comment]
Also cc @Expensify/design @dubielzyk-expensify - is this the correct padding around the "Welcome" message below the video?
CleanShot 2024-04-24 at 07 57 07@2x

**Comment by rezkiy37** [comment]
> > > I tested this case. The backend does not accept completing by clicking on task reports. The app passed all IDs correctly. I can complete tasks, once I open task reports. I am not sure that it is a frontend bug. The backend successfully opens task reports, but not complete. Is it possible that the backend does not see task reports until I open it?
> > 
> > 
> > On this, @thienlnam's explanation [in thread](https://expensify.slack.com/archives/C036QM0SLJK/p1712961069852729?thread_ts=1712620115.124819&cid=C036QM0SLJK) is that until you open a task, you aren't granted ‘read, write, share’ permissions. So if you try to complete the task from the "preview" in the parent chat before you've opened the taskReport up, it fails due to a lack of permissions.
> > It sounds like we need to fix this on the backend then. It's pretty strange to show a preview and allow an action to be taken within said preview that will always result in an error. You can take actions to like [Approve]/[Pay] an expenseReport from it's preview component in the parent chat without having to open the expenseReport first, so I think we should bring taskReports in-line to allow the same. @thienlnam will you progress this?
> 
> Follow-up on this. It looks like for a task in a "normal" DM, we don't let the user get that far until they've clicked into the taskReport first. We disable the action from being taken until then:
> 
> image
> So another option as an interim step might be to do the same in the Concierge DM until we figure out a broader plan.

This behavior happens because the receiver does not have a task report object. There is a [check](https://github.com/Expensify/App/blob/08b7ff41656a34373761fb2bd505cc0dfa64be7b/src/libs/actions/Task.ts#L1006) for this. So it is impossible for the onboarding tasks because the current user generates them on their own side.

**Review by pecanoro — APPROVED** [review]

**Comment by thienlnam** [comment]
I can spin up a PR to have it shared within DM reports. Created an issue to track it here https://github.com/Expensify/Expensify/issues/390483

**Review by mountiny — APPROVED** [review]
Looks good, I would make sure to merge this before we deploy again

**Comment by mountiny** [comment]
So I am going to merge this one now. Francois has a fix for the thumbnails not working as expected [here](https://github.com/Expensify/App/pull/40924).

This PR is adding the correct thumbnails https://github.com/Expensify/App/pull/40933

And finally @rezkiy37 got a PR for the videos

File: src/libs/Navigation/getOnboardingModalScreenOptions/index.native.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import getRootNavigatorScreenOptions from '@libs/Navigation/AppNavigator/getRootNavigatorScreenOptions';
+import type {ThemeStyles} from '@styles/index';
+import type {StyleUtilsType} from '@styles/utils';
+
+function getOnboardingModalScreenOptions(isSmallScreenWidth: boolean, styles: ThemeStyles, StyleUtils: StyleUtilsType) {
+    return getRootNavigatorScreenOptions(isSmallScreenWidth, styles, StyleUtils).fullScreen;
```
[the remark this answers is not in the document]
Comment (rezkiy37) [reply src/libs/Navigation/getOnboardingModalScreenOptions/index.native.ts:6]: @parasharrajat sorry, I didn't get what you meant. What is the problem with have with the gestures?

**Pull request merged**